Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php

```
@@ -133,19 +135,28 @@ public function onBeforeEvent(EventInterface $eventInstance, EventEnvelope $even
             return;
         }
 
+        if ($eventInstance instanceof WorkspaceWasPartiallyDiscarded) {
+            foreach ($eventInstance->discardedNodes as $discardedNode) {
+                $this->scheduleCacheFlushJobForNodeAggregate(
+                    $this->contentRepository,
+                    $eventInstance->workspaceName,
+                    $discardedNode->nodeAggregateId
+                );
+            }
+        }
+
         if (
             $eventInstance instanceof NodeAggregateWasRemoved
             // NOTE: when moving a node, we need to clear the cache not just after the move was completed,
             // but also on the original location. Otherwise, we have the problem that the cache is not
             // cleared, leading to presumably duplicate nodes in the UI.
             || $eventInstance instanceof NodeAggregateWasMoved
         ) {
-            $workspace = $this->contentRepository->getWorkspaceFinder()->findOneByCurrentContentStreamId($eventInstance->getContentStreamId());
-            if ($workspace === null) {
+            try {
+                $contentGraph = $this->contentRepository->getContentGraph($eventInstance->workspaceName);
```

using `$eventInstance->workspaceName` fixes most of the ui discard failures